packages/aws-cdk-lib/aws-appsync/lib/channel-namespace.ts

```diff
@@ -0,0 +1,122 @@
+import { Construct } from 'constructs';
+import { IApi } from './api-base';
+import { CfnChannelNamespace } from './appsync.generated';
+import { AuthorizationType } from './auth-config';
+import { Code } from './code';
+import { IResource, Lazy, Names, Resource, Token } from '../../core';
+
+/**
+ * An AppSync channel namespace
+ */
+export interface IChannelNamespace extends IResource {
+  /**
+   * The ARN of the AppSync channel namespace
+   *
+   * @attribute
+   */
+  readonly channelNamespaceArn: string;
+}
+
+/**
+ * Options for an AppSync channel namespace
+ */
+export interface ChannelNamespaceOptions {
+  /**
+   * The name of the channel namespace.
+   * This name must be unique within the Api.
+   *
+   * @default - A name is automatically generated
+   */
+  readonly channelNamespaceName?: string;
+
+  /**
+   * The event handler function code
+   *
+   * @default - no code is used
+   */
+  readonly code?: Code;
+
+  /**
+   * The authorization mode to use for publishing messages on the channel namespace.
+   * This configuration overrides the default Authorization configuration.
+   *
+   * @default - not override the default Authorization configuration
+   */
+  readonly publishAuthModes?: AuthorizationType[];
+
+  /**
+   * The authorization mode to use for subscribing to messages on the channel namespace.
+   * This configuration overrides the default Authorization configuration.
+   *
+   * @default - not override the default Authorization configuration
+   */
+  readonly subscribeAuthModes?: AuthorizationType[];
+}
+
+/**
+ * Properties for an AppSync channel namespace
+ */
+export interface ChannelNamespaceProps extends ChannelNamespaceOptions {
+  /**
+   * The Api
+   */
+  readonly api: IApi;
+}
+
+/**
+ * An AppSync channel namespace
+ *
+ * @resource AWS::AppSync::ChannelNamespace
+ */
+export class ChannelNamespace extends Resource implements IChannelNamespace {
+  /**
+   * Use an existing channel namespace by ARN
+   */
+  public static fromChannelNamespaceArn(scope: Construct, id: string, channelNamespaceArn: string): IChannelNamespace {
+    class Import extends Resource implements IChannelNamespace {
+      public readonly channelNamespaceArn = channelNamespaceArn;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of the AppSync channel namespace
+   */
+  public readonly channelNamespaceArn: string;
+
+  constructor(scope: Construct, id: string, props: ChannelNamespaceProps) {
+    if (props.channelNamespaceName !== undefined && !Token.isUnresolved(props.channelNamespaceName)) {
```

Comment:
`ChannelNamespaceName`'s pattern is quoted from Management Console.
![image](https://github.com/user-attachments/assets/ccbc45b7-30dd-4ec8-ab3c-c4396f9390de)

Currently, this pattern is inconsistent with the one described in the CloudFormation documentation.

I've reported it. https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/2208